### PR TITLE
176 remove log config

### LINKED ISSUE #176: Deprecate log config
logging.cfg causes a number of problems, not least that it doesn't seem to work and gives unexpected results if it's included without realising it. We should deprecate it and any related functionality (at least part of logging_helper and how it's handled in the executables).

### PR DESCRIPTION
**Pull request by Letizia97**
Resolves #176

### CODE REVIEW

File: bin/receiver.py [hunk 1]
```diff
@@ -38,14 +38,11 @@ def main():
     op = OptionParser(description=__doc__, version=ver)
     op.add_option('-c', '--config', help='location of config file',
                   default='/etc/apel/receiver.cfg')
-    op.add_option('-l', '--log_config',
-                  help='location of logging config file (optional)',
```
Comment (tofu-rocketry) [inline bin/receiver.py:42]: Add "DEPRECATED" to start of help message.

File: bin/receiver.py [hunk 2]
```diff
@@ -59,7 +56,7 @@ def main():
         print('Cannot start SSM.  Pidfile %s already exists.' % pidfile)
         sys.exit(1)
 
-    ssm.agents.logging_helper(cp, options.log_config)
```
Comment (tofu-rocketry) [inline bin/receiver.py:62]: Restore this line as we'll need to pass `log_config` on for now.

File: bin/sender.py [hunk 3]
```diff
@@ -36,15 +36,13 @@ def main():
     op = OptionParser(description=__doc__, version=ver)
     op.add_option('-c', '--config', help='location of config file',
                   default='/etc/apel/sender.cfg')
-    op.add_option('-l', '--log_config',
-                        help='location of logging config file (optional)',
```
Comment (tofu-rocketry) [inline bin/sender.py:40]: Add "DEPRECATED" to start of help message.

File: ssm/agents.py [hunk 4]
```diff
@@ -53,15 +53,14 @@
 REFRESH_DNS = 600
 
 
-def logging_helper(cp, log_manual_path=''):
+def logging_helper(cp):
     """Take config parser object and set up root logger."""
     try:
-        if os.path.exists(log_manual_path):
```
Comment (tofu-rocketry) [inline ssm/agents.py:59]: Re-inlcude this line, but add `logging.warning` message, but also note that it needs to be after line 63 where the logging is set up.

**Review by tofu-rocketry — CHANGES_REQUESTED** [review]

**Review by tofu-rocketry — APPROVED** [review]

File: bin/receiver.py (hunk 2 above)
Comment (gregcorbett) [reply bin/receiver.py:62]: does this line need to be restored? It looks like it wasn't but the conversation was resolved.

File: bin/sender.py [hunk 5]
```diff
@@ -37,14 +38,20 @@ def main():
...
+
+    # Deprecating functionality.
+    old_log_config_default_path = '/etc/apel/logging.cfg'
+    if (os.path.exists(old_log_config_default_path) or options.log_config is not None):
+        logging.warning('Separate logging config file option has been deprecated.')
 
     cp = ConfigParser.ConfigParser({'use_ssl': 'true'})
     cp.read(options.config)
 
-    ssm.agents.logging_helper(cp, options.log_config)
```
Comment (gregcorbett) [inline bin/sender.py:47]: As with the receiver, does this line need to be restored?

Comment (tofu-rocketry) [reply bin/sender.py:47]: Path check added to sender.py and receiver.py so we don't need to pass through the path in log_config.

File: bin/receiver.py (hunk 2 above)
Comment (tofu-rocketry) [reply bin/receiver.py:62]: Path check added to sender.py and receiver.py so we don't need to pass through the path in log_config.

**Pull request merged**